src/chains/ethereum/ethereum/src/transaction-pool.ts

```diff
@@ -452,6 +453,37 @@ export default class TransactionPool extends Emittery<{ drain: undefined }> {
     }
     return null;
   }
+  /**
+   * Deep clones and resets the transaction pool's `executables` such that `inProgress`
+   * transactions are moved back to `pending` and all transactions are unlocked.
+   * @returns Cloned and reset copy of transaction pool's `executables`.
+   */
+  public cloneAndResetExecutables() {
+    const executables = cloneDeep(this.executables);
```

Comment:
I like it! I have a couple of follow up questions.
1. Do you think adding this before #1330 will further entrench us in tech debt in our transaction classes?
  - If so, do you think it's worth making a separate issue for this and linking it to #1330 to be worked in conjunction?
2. Since we add all `inProgress` transactions to `pending`, wouldn't it be more performant to look through `pending` transactions first and then the `inProgress` transactions? It seems like we're doing extra work, unless I'm missing something.
3. Why do spell checkers think that performant is not a word? I feel like they're gas lighting me when they underline that word in red.

Reply:
3. I Google "performant" every time. I've finally done `Right Click` -> `Add to dictionary`. Thanks!
2. My _hunch_ is that `inProgress` first would be more performant because the transactions in `inProgress` will always sort _before_ the transactions in `pending`, and insertion queues (`Heap`) is fastest when the items being inserted are already ordered (I have NOT actually tested this though).
1. It might be adding some tech debt, but I'm worried that `deepClone` has hidden/subtle bugs, since we are cloning class instances, not just plain objects -- and hidden/subtle bugs probably worse.

Reply:
@davidmurdoch Thanks for the response. Your backwards ordering is deeply concerning.